### PR TITLE
Mark packages to be deprecated

### LINKED ISSUE #442: Deprecate unnecessary packages
We have started the process of deprecating packages that we no longer plan to use for the Xmidt project going forward.

The packages are:

- [x] `concurrent`

- [x] `conlimiter`
- [x] `health`
- [x]  `httppool`
- [x]  `logging`
- [x]  `secure`
- [x] `server`
- [x] `service`
- [x] `store`
- [x] `types`
- [x] `xerrors`
- [x]  `xmetrics`
- [x] `xviper`

### PR DESCRIPTION
**Pull request by joe94**
This PR marks packages the team has agreed on deprecating.

Closes #442

### CODE REVIEW

File: concurrent/doc.go [hunk 1]
```diff
@@ -1,5 +1,9 @@
 /*
 Package concurrent provides common functionality for dealing with concurrency that extends
 or enhances the core golang packages.
+
+Deprecated: concurrent is no longer planned to be used by future WebPA/XMiDT services
```
Comment (kristinaspring) [inline concurrent/doc.go]:
```suggestion
Deprecated: concurrent is no longer planned to be used by future WebPA/XMiDT services.
```

File: concurrent/doc.go [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+
+This package is frozen and no new functionality will be addded.
```
Comment (kristinaspring) [inline concurrent/doc.go]:
```suggestion
This package is frozen and no new functionality will be added.
```

File: conlimiter/doc.go [hunk 3]
```diff
@@ -1,5 +1,9 @@
 /*
 Package conlimiter provides a simple connection limiter for incoming TCP
 connections.
+
+Deprecated: conlimiter is no longer planned to be used by future WebPA/XMiDT services
```
Comment (kristinaspring) [inline conlimiter/doc.go]:
```suggestion
Deprecated: conlimiter is no longer planned to be used by future WebPA/XMiDT services.
```

File: conlimiter/doc.go [hunk 4]
```diff
@@ -1,5 +1,9 @@
...
+
+This package is frozen and no new functionality will be addded.
```
Comment (kristinaspring) [inline conlimiter/doc.go]:
```suggestion
This package is frozen and no new functionality will be added.
```

File: health/doc.go [hunk 5]
```diff
@@ -0,0 +1,8 @@
+/*
+Package health provides a simple heartbeat strategy for WebPA/XMiDT services
```
Comment (kristinaspring) [inline health/doc.go]:
```suggestion
Package health provides a simple heartbeat strategy for WebPA/XMiDT services.
```

File: health/doc.go [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+
+Deprecated: health is no longer planned to be used by future WebPA/XMiDT services
```
Comment (kristinaspring) [inline health/doc.go]:
```suggestion
Deprecated: health is no longer planned to be used by future WebPA/XMiDT services.
```

File: health/doc.go [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+
+This package is frozen and no new functionality will be addded.
```
Comment (kristinaspring) [inline health/doc.go]:
```suggestion
This package is frozen and no new functionality will be added.
```

File: httppool/doc.go [hunk 8]
```diff
@@ -1,5 +1,9 @@
 /*
 Package httppool provides a simple, configurable worker pool for dispatching HTTP transactions
 to servers.
+
+Deprecated: httppool is no longer planned to be used by future WebPA/XMiDT services
```
Comment (kristinaspring) [inline httppool/doc.go]:
```suggestion
Deprecated: httppool is no longer planned to be used by future WebPA/XMiDT services.
```

File: httppool/doc.go [hunk 9]
```diff
@@ -1,5 +1,9 @@
...
+
+This package is frozen and no new functionality will be addded.
```
Comment (kristinaspring) [inline httppool/doc.go]:
```suggestion
This package is frozen and no new functionality will be added.
```

File: logging/doc.go [hunk 10]
```diff
@@ -1,4 +1,8 @@
 /*
 Package logging provides basic configurability for the go-kit/log packages.
+
+Deprecated: logging is no longer planned to be used by future WebPA/XMiDT services
```
Comment (kristinaspring) [inline logging/doc.go]:
```suggestion
Deprecated: logging is no longer planned to be used by future WebPA/XMiDT services.
```

File: logging/doc.go [hunk 11]
```diff
@@ -1,4 +1,8 @@
...
+
+This package is frozen and no new functionality will be addded.
```
Comment (kristinaspring) [inline logging/doc.go]:
```suggestion
This package is frozen and no new functionality will be added.
```

File: secure/doc.go [hunk 12]
```diff
@@ -1,4 +1,8 @@
 /*
 Package secure contains the security types common to WebPA applications
+
+Deprecated: secure is no longer planned to be used by future WebPA/XMiDT services
```
Comment (kristinaspring) [inline secure/doc.go]:
```suggestion
Deprecated: secure is no longer planned to be used by future WebPA/XMiDT services.
```

File: secure/doc.go [hunk 13]
```diff
@@ -1,4 +1,8 @@
...
+
+This package is frozen and no new functionality will be addded.
```
Comment (kristinaspring) [inline secure/doc.go]:
```suggestion
This package is frozen and no new functionality will be added.
```

File: server/doc.go [hunk 14]
```diff
@@ -1,4 +1,8 @@
 /*
 Package server provides the standard approach to executing servers for WebPA
+
+Deprecated: server is no longer planned to be used by future WebPA/XMiDT services
```
Comment (kristinaspring) [inline server/doc.go]:
```suggestion
Deprecated: server is no longer planned to be used by future WebPA/XMiDT services.
```

File: server/doc.go [hunk 15]
```diff
@@ -1,4 +1,8 @@
...
+
+This package is frozen and no new functionality will be addded.
```
Comment (kristinaspring) [inline server/doc.go]:
```suggestion
This package is frozen and no new functionality will be added.
```

File: service/doc.go [hunk 16]
```diff
@@ -1,4 +1,8 @@
 /*
 Package service provides basic integration with go.serversets
+
+Deprecated: service is no longer planned to be used by future WebPA/XMiDT services
```
Comment (kristinaspring) [inline service/doc.go]:
```suggestion
Deprecated: service is no longer planned to be used by future WebPA/XMiDT services.
```

File: service/doc.go [hunk 17]
```diff
@@ -1,4 +1,8 @@
...
+
+This package is frozen and no new functionality will be addded.
```
Comment (kristinaspring) [inline service/doc.go]:
```suggestion
This package is frozen and no new functionality will be added.
```

File: store/doc.go [hunk 18]
```diff
@@ -1,5 +1,9 @@
 /*
 Package store implements some additional atomic value storage on top of sync/atomic.
 In particular, this means transparent caching of arbitrary values.
+
+Deprecated: store is no longer planned to be used by future WebPA/XMiDT services
```
Comment (kristinaspring) [inline store/doc.go]:
```suggestion
Deprecated: store is no longer planned to be used by future WebPA/XMiDT services.
```

File: store/doc.go [hunk 19]
```diff
@@ -1,5 +1,9 @@
...
+
+This package is frozen and no new functionality will be addded.
```
Comment (kristinaspring) [inline store/doc.go]:
```suggestion
This package is frozen and no new functionality will be added.
```

File: types/doc.go [hunk 20]
```diff
@@ -1,5 +1,9 @@
 /*
 Package types contains useful custom types that are common across the WebPA application.
 In particular, better JSON support for some of the more advance golang types is provided.
+
+Deprecated: types is no longer planned to be used by future WebPA/XMiDT services
```
Comment (kristinaspring) [inline types/doc.go]:
```suggestion
Deprecated: types is no longer planned to be used by future WebPA/XMiDT services.
```

File: types/doc.go [hunk 21]
```diff
@@ -1,5 +1,9 @@
...
+
+This package is frozen and no new functionality will be addded.
```
Comment (kristinaspring) [inline types/doc.go]:
```suggestion
This package is frozen and no new functionality will be added.
```

File: xerrors/doc.go [hunk 22]
```diff
@@ -0,0 +1,8 @@
+/*
+Package xerrors provides error unwrapping functionality
```
Comment (kristinaspring) [inline xerrors/doc.go]:
```suggestion
Package xerrors provides error unwrapping functionality.
```

File: xerrors/doc.go [hunk 23]
```diff
@@ -0,0 +1,8 @@
...
+
+Deprecated: xerrors is no longer planned to be used by future WebPA/XMiDT services
```
Comment (kristinaspring) [inline xerrors/doc.go]:
```suggestion
Deprecated: xerrors is no longer planned to be used by future WebPA/XMiDT services.
```

File: xerrors/doc.go [hunk 24]
```diff
@@ -0,0 +1,8 @@
...
+
+This package is frozen and no new functionality will be addded.
```
Comment (kristinaspring) [inline xerrors/doc.go]:
```suggestion
This package is frozen and no new functionality will be added.
```

File: xmetrics/doc.go [hunk 25]
```diff
@@ -1,3 +1,9 @@
-// Package xmetrics provides configurability for Prometheus-based metrics.  The more general go-kit interfaces
-// are used where possible.
+/*
+Package xmetrics provides configurability for Prometheus-based metrics.  The more general go-kit interfaces
+are used where possible.
+
+Deprecated: xmetrics is no longer planned to be used by future WebPA/XMiDT services
```
Comment (kristinaspring) [inline xmetrics/doc.go]:
```suggestion
Deprecated: xmetrics is no longer planned to be used by future WebPA/XMiDT services.
```

File: xmetrics/doc.go [hunk 26]
```diff
@@ -1,3 +1,9 @@
...
+
+This package is frozen and no new functionality will be addded.
```
Comment (kristinaspring) [inline xmetrics/doc.go]:
```suggestion
This package is frozen and no new functionality will be added.
```

File: xviper/doc.go [hunk 27]
```diff
@@ -0,0 +1,8 @@
+/*
+Package xviper provides customizations on use of viper for configuration loading
```
Comment (kristinaspring) [inline xviper/doc.go]:
```suggestion
Package xviper provides customizations on use of viper for configuration loading.
```

File: xviper/doc.go [hunk 28]
```diff
@@ -0,0 +1,8 @@
...
+
+Deprecated: xviper is no longer planned to be used by future WebPA/XMiDT services
```
Comment (kristinaspring) [inline xviper/doc.go]:
```suggestion
Deprecated: xviper is no longer planned to be used by future WebPA/XMiDT services.
```

File: xviper/doc.go [hunk 29]
```diff
@@ -0,0 +1,8 @@
...
+
+This package is frozen and no new functionality will be addded.
```
Comment (kristinaspring) [inline xviper/doc.go]:
```suggestion
This package is frozen and no new functionality will be added.
```

**Comment by joe94** [comment]
@kristinaspring, Thanks!

**Pull request merged**